spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/util/RandomLongSpanIdGenerator.java

```diff
@@ -0,0 +1,20 @@
+package org.springframework.cloud.sleuth.util;
+
+import java.util.Random;
+
+/**
+ * @author Marcin Grzejszczak
+ */
+public class RandomLongSpanIdGenerator implements SpanIdGenerator {
+
+	private Random random = new Random();
+
+	@Override
+	public long generateId() {
+		return random.nextLong();
+	}
+
+	public void setRandom(Random random) {
+		this.random = random != null ? random : new Random();
```

Comment:
super scary contract.. please remove this type and require a random to be supplied (and never changed)